### PR TITLE
Enhance rescale_array

### PR DESCRIPTION
**Pull request by Nic-Ma**
Follow up of #3235 

### Description
This PR added conversion for the output in rescale_array.

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick --unittests`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

File: monai/transforms/utils.py [hunk 1]
```diff
@@ -156,17 +156,26 @@ def rescale_array(
...
+        arr: input array to rescale.
+        minv: minimum value of target rescaled array.
+        maxv: maxmum value of target rescaled array.
+        dtype: data type of output, if None, same as input array.
+
     """
-    if dtype is not None:
-        arr, *_ = convert_data_type(arr, dtype=dtype)
+    dtype_ = dtype or arr.dtype
+    arr, *_ = convert_data_type(arr, dtype=dtype_)
```
Comment (ericspod) [inline monai/transforms/utils.py:168]: Is this doing something that the original didn't do? If `dtype` is `None` then doing no conversion seems fine, converting when `type_` is `arr.dtype` seems like a no-op?

Comment (Nic-Ma) [reply monai/transforms/utils.py]: Hi @ericspod ,

OK, sure, I changed the rescale_array back with more doc-strings for args, and updated `ScaleIntensity` to adjust the dtype in transform.
Could you please help take a look again?

Thanks in advance.

**Pull request merged**